buildtest/buildsystem/base.py

```diff
@@ -546,9 +546,9 @@ def _get_test_lines(self):
 
         if self.get_shell() in ["sh", "/bin/sh"]:
             lines += [f"#!/bin/sh"]
-            
+
         elif self.get_shell() in ["bash", "/bin/bash"]:
-            lines += [BUILDTEST_SHELL]
+            lines += [f"#!{BUILDTEST_SHELL}"]
 
```

Comment:
I think you would just want to do simply:

```bash
# Get the default shell executable
shell = which(self.get_shell())

# But if it doesn't exist, default to buildtest default
if not shell:
    shell = BUILDTEST_SHELL
lines += [f"#!{shell}"]
```
I don't see why we need all the custom logic for the shell type - we know it's one of bash or sh from the schema, and then it come down to finding an executable on the path. I'd even say that we should quit with an error if it's _not_ found because how would the script be run?

Reply:
there is no ``which`` method i can use either i use BuildTestCommand or we can leave as is. If we use the class something like, we are expecting `which` is installed in system which is generally not. I'd prefer hard coded path. or if you like a predefined shebangs in defaults.py.

I can look at this tomorrow or if you have a better alternative, feel free to merge into my PR. 
```
cmd = BuildTestCommand(f"which {self.get_shell}")
cmd.execute()
cmd.get_output()
```

Reply:
BuildTestCommand is already using which:

https://github.com/HPC-buildtest/buildtest-framework/blob/devel/buildtest/utils/command.py#L42

:)

shutil.which is supported after python 3. So, there is a which method you can use. And it wouldn't make sense to give to the build command, because it already uses it internally. Instead, you'd do as I showed you, a which() directly to the shell, which is what _the BuildTestCommand is already doing_...

Reply:
I already provided the exact code, if you want me to update the PR with it tomorrow I'd be happy to. Cheers!

Reply:
okay i think got it working as you mentioned. See https://github.com/HPC-buildtest/buildtest-framework/pull/229/commits/f88efa716344e93fd5c6936e9f82c5e2977d5e8f

I tested locally the shebangs were different but its okay some where ``/usr/bin/bash`` and some were ``/bin/bash``. It doesn't matter since recent distros including RHEL7 and Ubuntu have /bin symlinked to /usr/bin

```
gitpod /workspace/buildtest-framework $ ll -l /bi*
lrwxrwxrwx 1 root root 7 Jan 14 09:43 /bin -> usr/bin/
```

```
gitpod /workspace/buildtest-framework $ head -n 1 ~/.buildtest/testdir/{bzip2_help,bzip_compression,bzip_help,bzip_version}.sh 
==> /home/gitpod/.buildtest/testdir/bzip2_help.sh <==
#!/bin/bash

==> /home/gitpod/.buildtest/testdir/bzip_compression.sh <==
#!/usr/bin/bash

==> /home/gitpod/.buildtest/testdir/bzip_help.sh <==
#!/bin/sh

==> /home/gitpod/.buildtest/testdir/bzip_version.sh <==
#!/usr/bin/sh
```